myapp/app/controllers/tasks_controller.rb

```diff
@@ -15,6 +15,7 @@ def index
     sort_column = params[:sort].presence_in(Task.column_names) ? params[:sort] : "created_at"
     sort_direction = session[:is_order_desc] ? "DESC" : "ASC"
     @tasks = @tasks.order("#{sort_column} #{sort_direction}")
+    @tasks = @tasks.page(params[:page]).per(5)
```

Comment:
I think we should define the per-page value as a constant to avoid magic numbers.

Reply:
> Do you think it's better to allow user to filter tasks by both status and title?

yes

> Do you need @ to declare variable in search_title and filter_status

no maybe because Rails don't use it as an instance variable?

> instead of using class method, there any other ways?

I can directly write the logic in the controller

Reply:
> yes

Do you want to fix that?

> I can directly write the logic in the controller

You should write logic inside the model. I mean there any other implementation?. 

FYI Instead of using class methods, you can try using scope. With scope, you can use the function directly on the @tasks variable, so you don't need to call the function every time.

Reply:
> Do you want to fix that?

yes, I did.

> I will approve this PR after you fix the per_page path

I fixed this as well.


Now, I found the sorting by expiration_date function has a bug with pagination... (Every time, I go to the next page, the sorting also changes...😭)
I am fixing it on Monday.

Reply:
Have a nice weekend 👍

Reply:
I fixed the bug.